Database/MongoDB/Query.hs

```diff
@@ -130,6 +130,7 @@ data Failure =
     | WriteConcernFailure Int String  -- ^ Write concern error. It's reported only by insert, update, delete commands. Not by wire protocol.
     | DocNotFound Selection  -- ^ 'fetch' found no document matching selection
     | AggregateFailure String -- ^ 'aggregate' returned an error
+    | FindFailure String -- ^ 'find' returned an error
```

Comment:
Adding cases to types is not exactly backward compatible change. Is it possible to use QueryFailure here? findCommand seems to accept Query. It looks logical to use QueryFailure here.

Reply:
@VictorDenisov thanks for the review!
yeah it makes sense... fixed it here 951109c91d09807c0c32486e1815f9c001e036ae. Let me know what you think

Reply:
Looks good. I'll do another run over the code and will probably merge it closer to the weekend.